lib/charms/rolling_ops/v0/rollingops.py

```diff
@@ -0,0 +1,375 @@
+# Copyright 2022 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A library to enable charms to implement "rolling" or "serialized" operations.
+
+E.g., a rolling restart.
+
+You may use this library directly, or extend it to customize behavior.
+
+For example, in order to implement a rolling restart, a charm author would need to add the
+following lines of code, to the following files (note the consistent use of the name "restart")
+to namespace this particular rolling op:
+
+Add a peer relation to `metadata.yaml`:
+```yaml
+peers:
+    restart:
+        interface: rolling_op
+```
+
+Import the library, and enable it by doing initializing a RollingOpsManager class, passing
+in the Charm, name of the peer relation, and restart handler.
+
+src/charm.py
+```python
+# ...
+from charms.rolling_ops.v0.rollingops import RollingOpsManager, RollingEvents
```

Comment:
Whoops. This import line needs to be updated. (No need to import RollingEvents.)